doc/manual/src/release-notes/rl-next.md

```diff
@@ -4,6 +4,19 @@
   passed the `--stdin` flag. This is primarily useful when you have a large amount of paths which
   exceed the OS arg limit.
 
+* The `builtins.readDir` function has been optimized when encountering not-yet-known
+  file types from POSIX's `readdir`. In such cases the type of each file is/was
+  discovered by making multiple syscalls. This change makes these operations
+  lazy such that these lookups will only be performed if the attribute is used.
+  This optimization affects a minority of filesystems and operating systems.
+
```

Comment:
I think this was from a previous release and you accidentally kept it as part of the merge conflict.

Reply:
There was a merge conflict which I solved by "keeping 'em both".

Reply:
Yeah just keep the new one. The merge conflicts in this file are annoying and misleading, sad to say. We should adopt a different system in order to make this easier.